colcon_ros_cargo/task/ament_cargo/build.py

```diff
@@ -0,0 +1,180 @@
+# Licensed under the Apache License, Version 2.0
+
+import os
+import shutil
+from pathlib import Path
+
+from colcon_cargo.task.cargo import CARGO_EXECUTABLE
+from colcon_cargo.task.cargo.build import CargoBuildTask
+
+from colcon_core.logging import colcon_logger
+from colcon_core.plugin_system import satisfies_version
+from colcon_core.shell import create_environment_hook
+from colcon_core.task import TaskExtensionPoint
+from colcon_core.task import run
+
+import toml
+
+
+logger = colcon_logger.getChild(__name__)
+
+# Some logic needs to be executed once per run.
+# There are no colcon hooks for this, so it is shoehorned into the build step
+# with a global.
+package_paths = None
+
+
+class AmentCargoBuildTask(CargoBuildTask):
+    """A build task for packages with Cargo.toml + package.xml.
+
+    The primary problem that needs to be solved is that dependencies on other
+    packages in the same workspace are expressed by just a name in colcon, but
+    by a full path in Cargo.
+
+    That means, when building a Cargo package, all the packages it depends on
+    need to be resolved to full paths, and those need to be written into the
+    Cargo.toml, or alternatively into a [patch] section of a .cargo/config.toml
+    file. Here the latter approach is used.
+    """
+
+    def __init__(self):  # noqa: D107
+        super().__init__()
+        satisfies_version(TaskExtensionPoint.EXTENSION_POINT_VERSION, '^1.0')
+
+    def add_arguments(self, *, parser):  # noqa: D102
+        parser.add_argument(
+            '--lookup-in-workspace',
+            action='store_true',
+            help='Look up dependencies in the workspace directory. '
+            'By default, dependencies are looked up only in the installation '
+            'prefixes. This option is useful for setting up a '
+            '.cargo/config.toml for subsequent builds with cargo.')
```

Comment:
Does this mean that by default `.cargo/config.toml` is not written to the source workspace?

Reply:
No, it is, this option just affects the paths inside it.